### ISSUE TITLE
Gateway API originated LB service only contains a single port

### ISSUE THREAD

**Issue by rauanmayemir** [cilium/cilium#27013]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

For a gateway CRD with multiple listeners and different ports, Cilium will create an LB service with only one port. E.g for this:
```
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: http
  namespace: envoy-system
spec:
  gatewayClassName: cilium
  listeners:
    - protocol: HTTP
      port: 80
      name: http-ingress1
      hostname: "my.example.com"
      allowedRoutes:
        namespaces:
          from: All
    - protocol: HTTP
      port: 81
      name: http-ingress2
      hostname: "my.example.com"
      allowedRoutes:
        namespaces:
          from: All
```

there will only be a Service resource with port 81.

### Cilium Version

1.13.4, 1.14-rc0, 1.14-rc1

### Kernel Version

5.15.119-flatcar

### Kubernetes Version

1.26.5

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by youngnick** [comment]
Thanks for this issue @rauanmayemir, that does seem like a bug. I guess we will also need to look at adding a conformance test for this behavior upstream as well.

**Comment by rauanmayemir** [comment]
This was fixed by #27695
